metadata/upload.go

```diff
@@ -296,45 +374,37 @@ func parseRows(
 					inputs = append(inputs, kInputs...)
 				}
 
-				files := []*BQInput{}
 				nonFiles := []*BQInput{}
-
-				sizedFiles := make(chan *BQInput, len(inputs))
-				errs := make(chan error, 1)
-
-				for _, input := range inputs {
-					if input.Type == "file" {
-						go func(file BQInput) {
-							f := &file.Value
-							size, err := getGcsSize(ctx, f.StringVal)
-							if err != nil {
-								if getStatus(err) == http.StatusForbidden {
-									fmt.Println(err)
-									f.Valid = false
-								} else if strings.Contains(err.Error(), "doesn't exist") {
-									f.Valid = false
-								} else {
-									errs <- fmt.Errorf("%s: %s", f.StringVal, err.Error())
-									return
-								}
-							}
-							f.StringVal = fmt.Sprintf("%f", size)
-							sizedFiles <- &file
-						}(*input)
+				// create & stuff a channel holding jobs
+				fileCnt := 0
+				var jobs = make(chan GcsQueryJob, len(inputs))
+				for i, input := range inputs {
+					if (input.Type == "file") {
+						fileCnt++
+						job := GcsQueryJob{i, input}
+						jobs <- job
 					} else {
 						nonFiles = append(nonFiles, input)
 					}
 				}
+				close(jobs)
+				fmt.Fprintf(os.Stderr, "  Total number of inputs to go through: %d\n", len(inputs))
+				fmt.Fprintf(os.Stderr, "  Total number of files  to go through: %d\n", fileCnt)
+				fmt.Fprintf(os.Stderr, "  Total number of nonFiles            : %d\n", len(nonFiles))
```

Comment:
```suggestion
```
Having a log statement for every call results in excessive logging for a huge number of calls. I think it's better to remove it altogether.